sw/device/silicon_creator/manuf/skus/earlgrey_a0/generic/sram_cp_provision_init.c

```diff
@@ -44,16 +61,49 @@ static status_t peripheral_handles_init(void) {
   return OK_STATUS();
 }
 
+static status_t write_flash_info_fields(
+    manuf_cp_provisioning_data_t *provisioning_data) {
+  const flash_info_field_t fields[kNumFlashInfoFields] = {
+      kFlashInfoFieldDeviceId, kFlashInfoFieldManufState,
+      kFlashInfoFieldWaferAuthSecret};
+  const uint32_t *data[kNumFlashInfoFields] = {
+      provisioning_data->device_id, provisioning_data->manuf_state,
+      provisioning_data->wafer_auth_secret};
+  const uint32_t data_sizes[kNumFlashInfoFields] = {
+      kHwCfgDeviceIdSizeIn32BitWords, kHwCfgManufStateSizeIn32BitWords,
+      kFlashInfoWaferAuthSecretSizeIn32BitWords};
+  uint32_t byte_address;
+
+  for (size_t i = 0; i < ARRAYSIZE(fields); ++i) {
+    byte_address = 0;
+    TRY(flash_ctrl_testutils_info_region_setup(
+        &flash_ctrl_state, fields[i].page, fields[i].bank, fields[i].partition,
+        &byte_address));
+    // We skip the erasing of the ManufState field page as this field shares a
+    // page with the DeviceId field.
+    if (i == 1) {
```

Comment:
I must be missing something: for the first item (device_id), you are just writing, and for the second (manuf state), you are erasing and writing. This seems to be the opposite of what the comment says? Also won't be device ID be erased?

Reply:
The first and second items (device_id and manuf_state) are at indices 0 and 1, not 1 and 2. They happen to share a flash info page (page 0). So when we write device_id (at index 0), we erase and write the page, but when we write  the manuf_state, we just write to the page, as we don't want to erase what was already there. Lastly, the wafer authentication secret (at index 2) is on its own page, so that should also be an erase and write op.

Reply:
I updated the comment to make this more clear.